### PR TITLE
ENH add sample_weight to delinquency_curve

### LINKED ISSUE #11: ENH Add `sample_weight` to `delinquency_curve`
Right now, `delinquency_curve` is missing an important parameter: `sample_weight`.

It should not be that hard to implement it using some similar logic to `cap_curve`'s:
https://github.com/dihanster/datalib/blob/fcff6c2b09810984096dc8c76c4a8fff056747bc/datalib/metrics/_ranking.py#L60-L61

We can discuss this more if needed.

### PR DESCRIPTION
**Pull request by jeancmaia**
Hey peeps. In this PR, I update the delinquency curve to accept sample_weight. It is a pocket PR, and evaluating it shall not be a big deal. Thanks :) 

fixes #11

### CODE REVIEW

**Review by vitaliset — COMMENTED** [review]
Already looking great. I would suggest one new test: running with sample_weight=None should be the same as running with sample_weight = np.ones_like(y), and 5 * np.ones_like(y).

**Review by vitaliset — APPROVED** [review]
LGTM! Thanks for the PR!

**Review by vitaliset — APPROVED** [review]
Putting back my +1 here as CI is green now.

**Review by vitaliset — APPROVED** [review]
CI is green. We need the second reviewer.

File: datalib/metrics/_ranking.py [hunk 1]
```diff
@@ -135,12 +138,22 @@ def delinquency_curve(y_true, y_score, pos_label=None):
...
+    weights = weights / weights.max()
     y_true_sorted_by_scores = y_true[scores_idxs].copy()
     y_true_sorted = y_true[actual_idxs].copy()
+    sorted_weights = weights[scores_idxs].copy()
 
-    list_index = np.arange(1, len(y_true_sorted_by_scores) + 1)
-    approval_rate = np.append(0, list_index / len(list_index))
-    default_rate = np.append(0, y_true_sorted_by_scores.cumsum() / list_index)
-    optimal_rate = np.append(0, y_true_sorted.cumsum() / list_index)
+    step_approval_rate = 1 / len(y_true_sorted_by_scores)
```
Comment (isaac-tsousa) [inline datalib/metrics/_ranking.py:150]: Very elegant, very nice! Kudos

**Review by isaac-tsousa — APPROVED** [review]
LGTM; Great work

**Pull request merged**